stellar_resources/src/utils.rs

```diff
@@ -0,0 +1,26 @@
+use serde::{Deserialize, Deserializer};
+use serde::de;
+use std::str::FromStr;
+
+pub fn deserialize_from_str<'de, D, T>(d: D) -> Result<T, D::Error>
+where
+    D: Deserializer<'de>,
+    T: FromStr,
+{
+    // Call string deserialize on the "deserializer".
+    let s = String::deserialize(d)?;
+    // Now that we have a string, we can call FromStr
+    T::from_str(&s).map_err(|_| de::Error::custom("failed to parse string field"))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json::value::Value;
+
+    #[test]
+    fn it_parses_a_string_to_u64() {
+        let value = Value::String("123".to_string());
+        assert_eq!(deserialize_from_str::<Value, u64>(value).unwrap(), 123u64);
+    }
```

Comment:
What policy do we want to have on testing the error side of results?  I'm happy to go with the approach of letting the compiler enforce the error type and letting the calling functions worry about testing outcomes.  Shall we make this the agreed upon workflow?

Reply:
I was thinking about adding a failure case as well (like if `"abc"` comes in, since technically that "could" happen and we should make sure the outcome is correct (aka, doesn't panic).

I'd rather put the tests here than associated with the derived json struct.

Reply:
I leave it up to you.

Reply:
There is some logic, y'know? Like if it fails we return a particular error. I think it's probably important to include that as a test, and only test it here instead of all the places we include the derive. I'm adding a spec for it now.